app/models/bulk_download.rb

```diff
@@ -25,4 +30,161 @@ def convert_params_to_json
   def self.viewable(user)
     user.bulk_downloads
   end
+
+  def validate_access_token(access_token)
+    return self.access_token == access_token
+  end
+
+  def output_file_presigned_url
+    begin
+      return S3_PRESIGNER.presigned_url(:get_object,
+                                        bucket: ENV['SAMPLES_BUCKET_NAME'],
+                                        key: download_output_key,
+                                        expires_in: OUTPUT_DOWNLOAD_EXPIRATION).to_s
+    rescue StandardError => e
+      LogUtil.log_err_and_airbrake("BulkDownloadPresignError: #{e.inspect}")
+    end
+    nil
+  end
+
+  def server_host
+    if Rails.env == "prod"
+      "https://idseq.net"
+    elsif Rails.env == "staging"
+      "https://staging.idseq.net"
+    else
+      # You can use a tool like ngrok to test bulk downloads locally.
+      # DEVELOPMENT_TUNNEL_URL=https://abcdef.ngrok.io docker-compose up web.
+      ENV["DEVELOPMENT_TUNNEL_URL"]
+    end
+  end
+
+  # The Rails success url that the s3_tar_writer task can ping once it succeeds.
+  def success_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_success_path(access_token: access_token, id: id)}"
+  end
+
+  # The Rails error url that the s3_tar_writer task can ping if it fails.
+  def error_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_error_path(access_token: access_token, id: id)}"
+  end
+
+  # The Rails progress url that the s3_tar_writer task can ping to update progress.
+  def progress_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_progress_path(access_token: access_token, id: id)}"
+  end
+
+  def aegea_ecs_submit_command(
+    shell_command,
+    task_role: "idseq-downloads-#{Rails.env}",
+    ecr_image: "idseq-s3-tar-writer:latest",
+    fargate_cpu: 4096,
+    fargate_memory: 8192
+  )
+    command = "aegea ecs run --command=#{Shellwords.escape(shell_command)} "
+    command += "--task-role #{Shellwords.escape(task_role)} "
+    command += "--ecr-image #{Shellwords.escape(ecr_image)} "
+    command += "--fargate-cpu #{fargate_cpu} "
+    command += "--fargate-memory #{fargate_memory}"
+    command
```

Comment:
I can see you using `Shellwords.escape` here, but it is even safer to pass parameters individually to open3.

Ex:
```ruby
def aegea_ecs_submit_command(...)
    ["aegea", "ecs", "run",
     "--command", shell_command,
     "--ecr-image", ecr_image,
     "--fargate-cpu", fargate_cpu,
     "--fargate-memory", fargate_memory ]
end

...

  def kickoff_ecs_task(shell_command)
    aegea_command = aegea_ecs_submit_command(shell_command)
    # notice that I'm using the splat operator (asterisk) here, because capture 3 wants separated arguments instead of an array
    command_stdout, command_stderr, status = Open3.capture3(*aegea_command)
```